enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/PrimitiveExecutionContext.scala

```diff
@@ -110,14 +118,78 @@ case class PrimitiveExecutionContext(slots: SlotConfiguration) extends Execution
 
   private def fail(): Nothing = throw new InternalException("Tried using a primitive context as a map")
 
-  // This method is called from ScopeExpressions. We should already have allocated a slot for the given
-  // key, so we just set the value in the existing slot instead of creating a new context like in
-  // the MapExecutionContext.
+  //-----------------------------------------------------------------------------------------------------------
+  // Compatibility implementations of the old ExecutionContext API used by Community interpreted runtime pipes
+  //-----------------------------------------------------------------------------------------------------------
+  override def get(key: String): Option[AnyValue] = {
+    slots.get(key) match {
+      case Some(LongSlot(offset, false, CTNode)) =>
```

Comment:
Yes in a later PR I have added this comment about a suggested improvement:
```
  // TODO: As an optimization we can create a map from string key to precompiled getter/setter methods in the SlotConfiguration and use below
  // to avoid matching and unapplys
```

Reply:
👍